### DISCUSSION TITLE
Publishing package with scope different from username

### DISCUSSION THREAD

**Discussion by xenobytezero** [community/community#52267] [Packages] [answered]
### Select Topic Area

Question

### Body

I've been trying for a while to publish a package from a Yarn monorepo to Github Packages associated with a private repo, and I'm running out of things to Google to try and fix my issue.

The main difference I'm seeing with the many tutorials out there is that the scope I'm trying to publish my package to (`@ges`) is different from my username (`xenobytezero`).

If I publish with the scope changed to `@xenobytezero` everything seems to work fine, but not when I use the scope I want.

Do I need to use organisations for this? is there no way I can publish with an arbitary scope for a private repo for private access?

**Comment by SMelidoni** [comment, marked as answer]
Hi there @xenobytezero 👋🏽

In GitHub Packages, the scope for a package should match the owner of the repository when publishing to a user account. In your case, since you're publishing to a private repository owned by `xenobytezero`, you should use the `@xenobytezero` scope for your package.

However, if you want to use an arbitrary scope (like `@ges`), you can create an organisation on GitHub with that name. This would allow you to publish packages under the `@ges` scope.

To create an organisation and set up the desired scope:

1. **Create a GitHub organisation**: Follow the [documentation on creating a new organisation](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/creating-a-new-organization-from-scratch) and choose the name that matches your desired scope (e.g., "ges").

2. **Transfer your repository**: Transfer your private repository to the new organisation by following the [documentation on transferring a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository).

3. **Update your package scope**: Update your `package.json` file to use the desired scope:
`"@ges/package-name"`

4. **Configure your .npmrc or .yarnrc file**: Update the registry configuration and authentication set up in your `.npmrc` or `.yarnrc` file to use the new organisation scope.

- For `.npmrc`, update the following line:

  ```
  @ges:registry=https://npm.pkg.github.com/
  ```

- For `.yarnrc.yml`, update the following lines:

  ```yaml
  npmScopes:
    ges:
      npmRegistryServer: "https://npm.pkg.github.com"
  ```

After completing these steps, you should be able to publish packages under the `@ges` scope using your organisation.

Please note that creating an organisation may require a [paid plan](https://docs.github.com/en/billing) to enable private repositories within the organisation.

Hope this helps!

**Reply by DL259** [reply]
Npm no longer lets you use capital letters when naming a project but the organization I work for uses capitals in the name of the org. Do you have any advice getting around this?

**Reply by dir** [reply]
@DL259 Given it's been over a year, I'm sure you're long past this issue. But, I was in a similar boat and realized that you just have to use the all lowercase version of your org name. 

I.e:
GitHub Org: `My-Organization123`
NPM Scope: `my-organization123`
Package: `@my-organization123/example-package`

**Comment by xenobytezero** [comment]
Yep, creating an organisation lets me do what I need to. Was just a little unclear in the documentation. Thanks for your response!
